### PR TITLE
Update rspec to recent version and fix deprecations

### PR DESCRIPTION
**Pull request by briancain**
This overrides #8458

### CODE REVIEW

**Comment by gildegoma** [comment]
@briancain excellent, I'll review it next week! Thank you so much for the updates!

**Comment by briancain** [comment]
Hello @gildegoma - @chrisroberts and I are planning on merging this pretty soon, so I have gone ahead and cherry-picked your commit for the ansible tests so that they can go in when this gets merged!

**Comment by gildegoma** [comment]
@briancain these are awesome news! Can't wait!!!

File: Gemfile [hunk 1]
```diff
@@ -2,8 +2,9 @@ source "https://rubygems.org"
 
 gemspec
 
-if File.exist?(File.expand_path("../../vagrant-spec", __FILE__))
-  gem 'vagrant-spec', path: "../vagrant-spec"
-else
-  gem 'vagrant-spec', git: "https://github.com/mitchellh/vagrant-spec.git"
-end
+# Disable until vagrant-spec rspec is pulled up
```
Comment (chrisroberts) [inline Gemfile]: Lets re-enable this section before merge.

**Comment by gildegoma** [comment]
🎉 Excellent! Thank you merging all this so quickly!!!

**Comment by briancain** [comment]
Yesss!

**Pull request merged**